src/store/models/account.ts

```diff
@@ -16,28 +22,55 @@ interface LoginArgs {
 }
 
 export default createModel<RootModel>()({
-  state: { address: null, unspent: null } as AccountState,
+  state: { address: null, unspent: null, txs: {} } as AccountState,
   reducers: {
     SET_ADDRESS: (state, address: string) => ({
       ...state,
       address,
     }),
+    SET_TXS: (state, address, txs) => ({
+      ...state,
+      txs: {
+        ...txs,
+        [address]: txs,
+      },
+    }),
+    ADD_NEW_TX: (state, transaction: TxType) => {
+      const parsedTx = parseSpendTx(transaction.newTx);
+      parsedTx.recepient = transaction.recepient;
+      state.parsedTxs.unshift(parsedTx);
```

Comment:
Generally it's a Bad Idea™ to mutate state in react/redux (ever). I realize the state is being spread into a new object immediately afterwards but this can still have unintended and hard-to-debug consequences. If convenience is an issue we should use the `dot-prop-immutable` lib.